Command line quiz

artsiom-bukhautsou opened the pull request.

Commit:
task has been added

Commit:
files closing has been added

Commit:
package name has been changed

kortelyov commented on functions/gameFunctions.go:
```diff
@@ -0,0 +1,100 @@
+package functions
+
+import (
+	. "../models"
```
no relative passes - only "github.com/{username}/{project}..."

kortelyov commented on functions/gameFunctions.go:
```diff
@@ -0,0 +1,100 @@
+	var numberCorrectAnswers int
+	var numberIncorrectAnswers int
+	go calculateExpressions(&numberCorrectAnswers, &numberIncorrectAnswers)
+	time.Sleep(time.Minute)
```
bad solution
I think better to use WaitGroup for flow control (and move time management into goroutine)

kortelyov commented on functions/gameFunctions.go:
```diff
@@ -0,0 +1,100 @@
+)
+
+func StartGame() {
+	var numberCorrectAnswers int
```
long names, better correct / incorrect

kortelyov commented on functions/gameFunctions.go:
```diff
@@ -0,0 +1,100 @@
+		panic(err)
+	}
+	jsonByteValue, _ := ioutil.ReadAll(jsonFile)
+	_ = json.Unmarshal(jsonByteValue, &expressions)
```
error check

kortelyov commented on functions/gameFunctions.go:
```diff
@@ -0,0 +1,100 @@
+
+func calculateExpressions(numberCorrectAnswers *int, numberIncorrectAnswers *int) {
+	expressions := getExpressions()
+	for _, expression := range expressions {
```
what about random order ?

kortelyov commented on runner.go:
```diff
@@ -0,0 +1,27 @@
+package main
```
file name for runners - main.go (convention)

kortelyov commented on runner.go:
```diff
@@ -0,0 +1,27 @@
+)
+
+func main() {
+	fmt.Printf("To make a choice:\n" +
```
what about infinity loop for logic ?
add new menu point like "exit"

kortelyov commented on runner.go:
```diff
@@ -0,0 +1,27 @@
+	case 1:
+		StartGame()
+	case 2:
+		ratingList := GetRatingList()
```
stop using global functions

Commit:
-runner.go has been renamed to main.go
-exit function has been added
-shuffle function has been added to calculateExpression func
-variables name have been simplified

Commit:
channels have been added

Commit:
-error handling has been updated
-import path has been updated
-init function has been added
